### PR TITLE
Added devcontainer.

### PR DESCRIPTION
**Pull request by seriva**
Not sure if we want to include this but might be useful for the poor people using Windows.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+ARG VARIANT="buster"
+FROM mcr.microsoft.com/vscode/devcontainers/base:0-${VARIANT}
+
+ENV TERRAFORM_VERSION 0.13.2
+ENV YQ_VERSION=3.3.4
+
+# Install Terraform, yq, make
+RUN wget https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip \
```
Comment (sk4zuzu) [inline .devcontainer/Dockerfile:8]: I think we could use just wget or just curl, but not really that important though :thinking:

File: .devcontainer/setup-env.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+echo 'export M_VERSION="dev"' >> ~/.bashrc
+echo 'export M_WORKDIR="$(pwd)/workdir"' >> ~/.bashrc
+echo 'export M_RESOURCES="$(pwd)/resources"' >> ~/.bashrc
+echo 'export M_SHARED="$(pwd)/shared"' >> ~/.bashrc
+cd $(pwd)/resources/terraform && terraform init
```
Comment (sk4zuzu) [inline .devcontainer/setup-env.sh:5]:
```suggestion
cd $(pwd)/resources/terraform && terraform init
```

**Review by sk4zuzu — DISMISSED** [review]
LGTM

**Comment by mkyc** [comment]
@atsikham can you have look at this PR if it helps with OS related issues?

**Comment by atsikham** [comment]
> @atsikham can you have look at this PR if it helps with OS related issues?

I don't think so. Make tasks that we use for modules contain `docker` execution, which is unavailable in devcontainer.

**Comment by seriva** [comment]
Yeah with WSL2 I also have no issues to build/test/develop it. Only reason I like this approach is that it kind of sets up all your tooling versions and you dont have to build the container every time you need to test.

**Comment by seriva** [comment]
And one more thing is that it also allows to use some nice plugins for VSCode for various things like bash, python, terraform or whatever the module uses.

**Review by sk4zuzu — APPROVED** [review]
LGTM

**Review by rafzei — DISMISSED** [review]
LGTM

**Comment by atsikham** [comment]
there is an unnecessary space in path - `examples/basic_flow _devcontainer/azure.mk.sample`

**Review by atsikham — CHANGES_REQUESTED** [review]
Need to remove space from `basic_flow _devcontainer` dir name

**Comment by seriva** [comment]
@atsikham Fixed the directory issue.

**Review by atsikham — APPROVED** [review]

**Review by sk4zuzu — APPROVED** [review]
LGTM

**Pull request merged**